partner_resource_delivery_schedule/models/sale_order_line.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2023 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _prepare_procurement_values(self, group_id=False):
+        """Set the date_deadline and date_planned to the expected_date
+        of the SO if commitment_date is not set."""
+        self.ensure_one()
+        values = super()._prepare_procurement_values(group_id)
+        if self.order_id.commitment_date:
+            return values
+        values["date_deadline"] = self.order_id.expected_date
```

Comment:
I'm wondering if these shouldn't go in another module...

Reply:
These lines are present in `sale_stock` module and this module depends on it.
https://github.com/odoo/odoo/blob/16.0/addons/sale_stock/models/sale_order_line.py#L233